CHANGES/2735.bugfix

```diff
@@ -0,0 +1,2 @@
+Fix recreation of modular snippet when missing.
+To apply this fix re-sync with ``optimize=False`` and republish.
```

Comment:
@dralley do we want to document this more e.g. in the docs?

Reply:
I don't think it's necessary.  As far as we know it only happens when the user does something silly anyway.

Reply:
Truthfully I'm not even sure this needs to go in the changelog for that reason.